includes/functions/functions_communications.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+use http\Exception\BadQueryStringException;
+
+/**
+ *
+ * Communication-related functions, such as for making CURL requests
+ *
+ * @copyright Copyright 2003-2025 Zen Cart Development Team
+ * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
+ * @version $Id:  New in v2.2.0 $
+ *
+ */
+
+  function zenDoCurlRequest(string $url, string $method = 'GET', string|array|null $payload = null, $encodePayloadArraysAsJson = false, $decodeJsonResponses = false): string|false
+  {
+      $base_UA_host = defined('HTTP_CATALOG_SERVER') ? HTTP_CATALOG_SERVER : HTTP_SERVER;
+      $referrer = $base_UA_host . DIR_WS_CATALOG;
+      $userAgent = empty($_SERVER['HTTP_USER_AGENT']) ? $base_UA_host . DIR_WS_CATALOG : $_SERVER['HTTP_USER_AGENT'];
+
+      if (is_array($payload)) {
+          if ($encodePayloadArraysAsJson) {
+              $payload = json_encode($payload);
+          } else {
+              $payload = http_build_query($payload);
+          }
+          if ($payload === false) {
+              throw new BadQueryStringException('Could not encode the provided array $payload.');
+          }
+      }
+
+      $ch = curl_init();
+      curl_setopt($ch, CURLOPT_URL, $url);
+      curl_setopt($ch, CURLOPT_VERBOSE, 0);
+      curl_setopt($ch, CURLOPT_HEADER, false);
+      curl_setopt($ch, CURLOPT_USERAGENT, $userAgent);
+      curl_setopt($ch, CURLOPT_REFERER, $referrer);
+      curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
+      curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+
+      if (!empty($payload) && strtoupper($method) === 'POST') {
+          curl_setopt($ch, CURLOPT_POST, true);
+          curl_setopt($ch, CURLOPT_POSTFIELDS, $payload);
+      }
```

Comment:
Why are the CURL options 'hard-coded' instead of accepting an array of to-be-applied ones using `curl_setopt_array`?  Those hard-coded ones could be the default if the caller supplies an empty array of options.

Reply:
Updated.